tests/entrypoint.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2018 Red Hat, Inc.
+#
+
+set -e
+
+if [ -z "${KUBECONFIG}" ]; then
+    echo "KUBECONFIG is not provided"
+    exit 1
+fi
+ls -la "${WORKSPACE}"
+ls -la "${DATA_DIR}"
+if [ ! -e "${KUBECONFIG}" ] ; then
+    echo "KUBECONFIG=${KUBECONFIG} doesn't exist"
+    exit 1
+fi
+
+mkdir -p "${RESULTS_DIR}"
+
+# Make sure that test's requiremets are met
+oc apply -f ${WORKSPACE}/manifests/*.yaml
```

Comment:
Instead of expecting that the test-infra manifests are mounted in, I would simply copy them in.

Here how I expect the docker container build flow:
 * Copy in the templates of the testing-infra manifests in manifests/testing. 
 * Copy in the `manifest-templator` from `tools` into the container too.
 * Harmonize important flags for the test binaries and the maniefst-temlator (you can simply ignore unknown flags in the templator), so that we can simply call the templator during the entrypoint exectuion
 * Add the possibility to deploy the testing infrastructure from the test binary itself (to avoid the need for oc or kubectl)
 * It is now possible in openshift 3.9 to provide security contexts as yaml, we  can simply add it to manifests/testing (and ignore it for k8s)

So the final line of the entrypoint.sh should then look like `./tests.test "$@"` and we would simply execute the tests as follows:

Instead of 

```bash
tests.test -kubeconfig=${kubeconfig} -tag=${docker_tag} -prefix=${functest_docker_prefix} -kubectl-path=${kubectl} -test.timeout 60m
```

we would execute it as

```bash
docker run -v $HOME/.kube/config:/home/containeruser/.kube/config:z --rm kubevirt.io/tests:latest --docker-tag=latest --docker-prefix=docker.io/kubevirt --test.timeout 60m --apply-testing-infra --cluster-type=ocp
```

Reply:
Also things like

```bash
docker run --rm kubevirt.io/tests:latest --help
```

may work then.